doc/source/commands/files.rst

```diff
@@ -5,9 +5,25 @@ File I/O
 
 For information about where files are kept and how to deal with volumes see the :ref:`Volumes <volumes>` page in the general topics section of this documentation.
 
+.. contents::
+    :local:
+    :depth: 2
+
 .. note::
 
-    All file names (program names) must be valid Identifiers. They can not contain spaces or special characters. For example, you can't have a file name called "this is my-file".
+    *Limitations on file names used for programs*
+
+    All file names used as program names with the ``run`` command must be
+    valid identifiers.  They can not contain spaces or special characters. For
+    example, you can't have a program named ``this is my-file.ks``.  This rule
+    does not necessarily apply to other filenames such as log files.  However
+    to use a filename that contains spaces, you will have to put quotes around
+    it.
+
+    On case-sensitive filesystems typically found on Linux and Mac, you should
+    name program files used with the ``run`` command entirely with
+    lowercase-only filenames or the system may fail to find them when you
+    use the ``run`` command.
```

Comment:
Not a formatting change, but the prev text was wrong because it claimed ALL files followed this rule when really only program files used with `run` do.  I noticed this as I was trying to fix the ..note formatting.